plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1391,6 +1442,92 @@ private List<ColumnMetadata> getColumnMetadatas(Schema schema)
         return columns.build();
     }
 
+    @Override
+    public ConnectorAnalyzeMetadata getStatisticsCollectionMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Object> analyzeProperties)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        checkArgument(handle.getTableType() == DATA, "Cannot analyze non-DATA table: %s", handle.getTableType());
+
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, handle);
+        Set<String> allColumnNames = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(ColumnMetadata::getName)
+                .collect(toImmutableSet());
+
+        Set<String> analyzeColumnNames = getColumnNames(analyzeProperties)
+                .map(columnNames -> {
+                    // validate that proper column names are passed via `columns` analyze property
+                    if (columnNames.isEmpty()) {
+                        throw new TrinoException(INVALID_ANALYZE_PROPERTY, "Cannot specify empty list of columns for analysis");
+                    }
+                    if (!allColumnNames.containsAll(columnNames)) {
+                        throw new TrinoException(
+                                INVALID_ANALYZE_PROPERTY,
+                                format("Invalid columns specified for analysis: %s", Sets.difference(columnNames, allColumnNames)));
+                    }
+                    return columnNames;
+                })
+                .orElse(allColumnNames);
+
+        Set<ColumnStatisticMetadata> columnStatistics = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .filter(column -> analyzeColumnNames.contains(column.getName()))
```

Comment:
Do you need to lowercase `analyzeColumnNames` ?

Reply:
The user needs to provide them in lowercase. This makes it easier to support non-lowercase column names in the future.